base_geoengine/static/src/js/views/geoengine/layers_panel/layers_panel.esm.js

```diff
@@ -0,0 +1,241 @@
+        switch (action) {
+            case "onDomainChanged":
+                Object.assign(vectorLayer, {
+                    model_domain: value,
+                    onDomainChanged: true,
+                });
+                break;
+            case "onVisibleChanged":
+                Object.assign(vectorLayer, {isVisible: value, onVisibleChanged: true});
+                break;
+            case "onLayerChanged":
+                const geo_field_id = await this.orm.call(
+                    vectorLayer.resModel,
+                    "set_field_real_name",
+                    [value.geo_field_id]
+                );
+                const attribute_field_id = await this.orm.call(
+                    vectorLayer.resModel,
+                    "set_field_real_name",
+                    [value.attribute_field_id]
+                );
+                value.geo_field_id = geo_field_id;
+                value.attribute_field_id = attribute_field_id;
+                Object.assign(vectorLayer, {...value, onLayerChanged: true});
+                break;
+            case "onSequenceChanged":
+                if (vectorLayer !== undefined) {
+                    Object.assign(vectorLayer, {
+                        sequence: value,
+                        onSequenceChanged: true,
+                    });
+                }
+                break;
+        }
+    }
+
+    onEditFilterButtonSelected(vector) {
+        this.addDialog(DomainSelectorGeoFieldDialog, {
+            resModel: vector.model,
+            initialValue: vector.model_domain,
+            readonly: false,
+            isDebugMode: Boolean(this.env.debug),
+            model: vector,
+            onSelected: (value) => this.onEditFilterDomainChanged(vector, value),
+            title: this.env._t("Domain editing"),
+        });
+    }
+
+    async onEditFilterDomainChanged(vector, value) {
+        if (this.isGeoengineAdmin) {
+            const record = this.props.vectorModel.records.find(
+                (el) => el.resId === vector.resId
+            );
+            await record.update({model_domain: value});
+            await record.save();
+        }
+        this.onVectorChange(vector, "onDomainChanged", value);
+    }
+
+    async onEditButtonSelected(vector) {
+        const view = await this.rpc("/web/action/load", {
+            action_id: "base_geoengine.geo_vector_geoengine_view_action",
+        });
+
+        this.addDialog(FormViewDialog, {
+            resModel: vector.resModel,
+            title: this.env._t("Editing vector layer"),
+            viewId: view.view_id[0],
+            resId: vector.resId,
+            onRecordSaved: (record) =>
+                this.onVectorChange(vector, "onLayerChanged", record.data),
+        });
+    }
+    /**
+     * This method allows you to open/close the panel.
+     */
+    fold() {
+        this.state.isFolded = !this.state.isFolded;
+    }
+}
+
+LayersPanel.template = "base_geoengine.LayersPanel";
```

Comment:
starting in odoo 16.1, we officially allow the use of `static` in component. I like that a lot, since it feels like the static props/template/components key belong to the top of the component code